include/dca/linalg/util/magma_queue.hpp

```diff
@@ -25,26 +25,25 @@ namespace linalg {
 namespace util {
 // dca::linalg::util::
 
-class MagmaQueue : public CudaStream {
+class MagmaQueue {
 public:
   MagmaQueue() {
     cublasCreate(&cublas_handle_);
     cusparseCreate(&cusparse_handle_);
     int device;
     cudaGetDevice(&device);
-    magma_queue_create_from_cuda(device, static_cast<cudaStream_t>(*this), cublas_handle_,
+    magma_queue_create_from_cuda(device, stream_, cublas_handle_,
                                  cusparse_handle_, &queue_);
   }
 
   MagmaQueue(const MagmaQueue& rhs) = delete;
   MagmaQueue& operator=(const MagmaQueue& rhs) = delete;
 
-  MagmaQueue(MagmaQueue&& rhs) noexcept : CudaStream(std::move(rhs)) {
+  MagmaQueue(MagmaQueue&& rhs) noexcept {
```

Comment:
This causes performance issue: stream_ is default initialized to a new stream, then the swap happens, and when rhs goes out of scope the stream is destroyed without ever being used.